src/redux/userLists.ts

```diff
@@ -21,16 +23,90 @@ const USER_LISTS_UPDATE_LISTS = 'userLists/USER_LISTS_UPDATE_LISTS';
 const USER_LISTS_CLEAR_STATE = 'userLists/USER_LISTS_CLEAR_STATE';
 const USER_LISTS_SET_SELECTED_LIST = 'userLists/USER_LISTS_SET_SELECTED_LIST';
 const FAVORITES_LIST_ID = 'favorites';
+
 // -- Actions --------------------------------------------------------------- //
-export const userListsLoadState = () => async (dispatch, getState) => {
+
+/**
+ * A single user list, based on the data found in `DefaultTokenLists`.
+ */
+export interface UserList {
+  emoji: string;
+  id: string;
+  name: string;
+  tokens: string[];
+}
+
+/**
+ * The current `userLists` state.
+ */
+interface UserListsState {
+  lists: UserList[];
+  loadingUserLists: boolean;
+  ready: boolean;
+  selectedList: string | null;
+}
+
+/**
+ * A `userLists` Redux action.
+ */
+type UserListsAction =
+  | UserListsLoadRequestAction
+  | UserListsLoadSuccessAction
+  | UserListsSetSelectedListAction
+  | UserListsUpdateListsAction
+  | UserListsFailureAction
+  | UserListsReadyAction
+  | UserListsClearStateAction;
+
+interface UserListsLoadRequestAction {
+  type: typeof USER_LISTS_LOAD_REQUEST;
+}
+
+interface UserListsLoadSuccessAction {
+  type: typeof USER_LISTS_LOAD_SUCCESS;
+  payload: {
+    lists: UserList[];
```

Comment:
this isn't part of your TS changes, but noticed we can improve this: the payload could be the `UserList[]` directly instead of being an object with `lists` inside of it. This also means updating line 115 and line 241 accordingly